src/resolver/resolver.zig

```diff
@@ -833,6 +823,25 @@ pub const Resolver = struct {
 
         if (import_path.len == 0) return .{ .not_found = {} };
 
+        if (r.opts.mark_builtins_as_external) {
+            if (strings.hasPrefixComptime(import_path, "node:") or
+                strings.hasPrefixComptime(import_path, "bun:") or
+                bun.JSC.HardcodedModule.Aliases.has(import_path, r.opts.target))
+            {
+                return .{
+                    .success = Result{
+                        .import_kind = kind,
+                        .path_pair = PathPair{
+                            .primary = Path.init(import_path),
+                        },
+                        .is_external = true,
+                        .module_type = .cjs,
```

Comment:
This should be either `.esm` or `.cjs` depending on what the input kind was

```suggestion
                        .module_type = switch (kind) {
                            .require, .require_resolve => .cjs,
                            else => .esm
                        },
```